src/db.c

```diff
@@ -483,6 +485,12 @@ int dbGenericDeleteWithDictIndex(serverDb *db, robj *key, int async, int flags,
             debugServerAssert(0 == kvstoreHashtableDelete(db->expires, dict_index, key->ptr));
         }
 
+        /* If deleting a hash object, untrack it from the volatile set if needed.
+         * Note: the volatile set may already be empty if this deletion comes from active expiry. */
```

Comment:
same comment. volatile set has nothing to do with this.
```suggestion
        /* If deleting a hash object, un-track it from the volatile items tracking only if it contains volatile items.
         * Note: the object may already contain no volatile items if this deletion comes from active expiry. */
```